pkg/controller/securityonboarding/securityonboarding_controller.go

```diff
@@ -1116,6 +1163,68 @@ func getIAMOnboardJob(instance *operatorv1alpha1.SecurityOnboarding, r *Reconcil
 	currentJob := &batchv1.Job{}
 	err := r.client.Get(context.TODO(), types.NamespacedName{Name: "iam-onboarding", Namespace: instance.Namespace}, currentJob)
 	if err == nil {
+		if currentJob.Status.Conditions != nil && currentJob.Status.Conditions[0].Type == "Failed" {
```

Comment:
I think we need to add proper exception handling here to make sure the code is not broken.
eg.
```
config, err := rest.InClusterConfig()
  if err != nil {
		  return config, err
	  }
```